src/tic-tac-toe.ts

```diff
@@ -40,35 +43,70 @@ class Game {
     return square === this.currentPlayer;
   };
 
-  private checkRowsForWin(): boolean {
-    return this.board.some(row => row.every(this.isCurrentPlayer));
+  private checkRowsForWin(): WinningVector | undefined {
+    for (let rowIndex = 0; rowIndex < this.boardSize; rowIndex++) {
+      const row = this.board[rowIndex];
+      const vector: WinningVector = [];
+      for (let colIndex = 0; colIndex < this.boardSize; colIndex++) {
+        const square = row[colIndex];
+        if (this.isCurrentPlayer(square)) {
+          vector.push({ row: rowIndex, col: colIndex });
+        }
+      }
+      if (vector.length === this.boardSize) {
+        return vector;
+      }
+    }
+    return;
   }
 
-  private checkColumnsForWin(): boolean {
+  private checkColumnsForWin(): WinningVector | undefined {
     const transposedBoard = this.eachIndex().map(i =>
       this.board.map(row => row[i])
     );
 
-    return transposedBoard.some(row => row.every(this.isCurrentPlayer));
+    for (let colIndex = 0; colIndex < this.boardSize; colIndex++) {
+      const column = transposedBoard[colIndex];
+      const vector: WinningVector = [];
+
+      for (let rowIndex = 0; rowIndex < this.boardSize; rowIndex++) {
+        const square = column[rowIndex];
+        if (this.isCurrentPlayer(square)) {
+          vector.push({ row: rowIndex, col: colIndex });
+        }
+      }
+      if (vector.length === this.boardSize) {
+        return vector;
+      }
+    }
+    return;
   }
 
-  private checkDiagonalsForWin(): boolean {
-    const diagonals = [
-      this.eachIndex().map(i => this.board[i][i]),
-      this.eachIndex().map(i => this.board[i][this.boardSize - 1 - i]),
+  private checkDiagonalsForWin(): WinningVector | undefined {
+    const diagonals: WinningVector[] = [
+      // check values from top left to bottom right (ie for 3x3 grid => [0, 0], [1, 1], [2, 2])
+      this.eachIndex().map(i => ({ row: i, col: i })),
```

Comment:
For me, if I'm writing out a comment like this, I'll typically ask "Can I convey the same information with a variable name?"